app/services/alerts/generate_and_save_alerts_and_benchmarks.rb

```diff
@@ -0,0 +1,79 @@
+module Alerts
+  class GenerateAndSaveAlertsAndBenchmarks
+    def initialize(school:, aggregate_school: nil, benchmark_result_generation_run: nil, framework_adapter: FrameworkAdapter)
+      @school = school
+      @aggregate_school = aggregate_school || AggregateSchoolService.new(school).aggregate_school
+      @benchmark_result_generation_run = benchmark_result_generation_run || BenchmarkResultGenerationRun.create!
+      @framework_adapter = framework_adapter
+    end
+
+    def perform
+      ActiveRecord::Base.transaction do
+        @alert_generation_run = AlertGenerationRun.create!(school: @school)
+        @benchmark_result_school_generation_run = BenchmarkResultSchoolGenerationRun.create!(school: @school, benchmark_result_generation_run: @benchmark_result_generation_run)
+
+        relevant_alert_types.each do |alert_type|
```

Comment:
Add special case for AlertImpendingHoliday to run it separately as an alert and a benchmark with different flags.

Reply:
if alert_type.class_name = AlertImpendingHoliday
          service = GenerateAlertTypeRunResult.new(
            school: @school,
            aggregate_school: @aggregate_school,
            alert_type: alert_type,
            use_max_meter_date_if_less_than_asof_date: alert_type.fuel_type.present? ? true : false
          )
          alert_type_run_result = service.perform(Time.zone.today)
          process_alert_type_run_result(alert_type_run_result)

          service = GenerateAlertTypeRunResult.new(
            school: @school,
            aggregate_school: @aggregate_school,
            alert_type: alert_type,
            use_max_meter_date_if_less_than_asof_date: true
          )
          alert_type_run_result = service.perform(Time.zone.today)
          process_benchmark_type_run_result(alert_type_run_result) if alert_type.benchmark == true
else
 ...current code
end

Reply:
:+1: done - tho needs double checking i have the correct logic for the separate processes for AlertImpendingHoliday